pkg/kubelet/cri/remote/remote_image.go

```diff
@@ -52,10 +57,24 @@ func NewRemoteImageService(endpoint string, connectionTimeout time.Duration) (in
 	ctx, cancel := context.WithTimeout(context.Background(), connectionTimeout)
 	defer cancel()
 
-	conn, err := grpc.DialContext(ctx, addr,
+	dialOpts := []grpc.DialOption{}
+	dialOpts = append(dialOpts,
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
 		grpc.WithContextDialer(dialer),
 		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(maxMsgSize)))
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletTracing) {
+		tracingOpts := []otelgrpc.Option{
+			otelgrpc.WithPropagators(tracing.Propagators()),
+			otelgrpc.WithTracerProvider(tp),
+		}
+		// Even if there is no TracerProvider, the otelgrpc still handles context propagation.
```

Comment:
I looked at couple places - it looks like context used for these calls is often a new context. Is the plan to change all the callsite to start using an incoming context?

Reply:
I'm not against this PR, I think it is a good step forward. I just want to make sure there are follow up work planned to improve traces correlation

Reply:
Yes, you are right. This  would only enable creation of individual traces for every request made to the runtime. For proper tracing support, we would need to instrument spans that could cover multiple gRPC calls. For example, instrumenting containerRuntime.SyncPod or even kubelet syncPod would cover multiple CRI calls which would produce more valuable traces. 

I am interested to work on the follow up tasks. The initial KEP#2832  mainly targeted tracing around http and grpc calls. And I don't see any existing issue or discussion around extending opentelemetry tracing support in kubelet for trace correlation. What will be the best way to start this discussion? Should I create an issue identifying the gaps in tracing in kubelet?

Reply:
issue is a good place to discuss it.

I understand the minimal step taken here

/lgtm
/unhold